lib/datadog/appsec/contrib/rack/request_middleware.rb

```diff
@@ -66,34 +66,30 @@ def call(env)
 
             return @app.call(env) unless ready
 
-            gateway_request = Gateway::Request.new(env)
-
             add_appsec_tags(processor, scope)
             add_request_tags(scope, env)
 
-            request_return, request_response = catch(::Datadog::AppSec::Ext::INTERRUPT) do
-              Instrumentation.gateway.push('rack.request', gateway_request) do
+            gateway_request = Gateway::Request.new(env)
+            http_response = nil
+            gateway_response = nil
```

Comment:
Just a matter of preference, but maybe this?

```suggestion
            http_response = nil
            gateway_request = Gateway::Request.new(env)
            gateway_response = nil
```

Reply:
fixed [b351e16](https://github.com/DataDog/dd-trace-rb/pull/4281/commits/b351e161989a58e738e95e5f4dce2beff5594451)